SparseGrids/tsgLinearSolvers.cpp

```diff
@@ -237,6 +256,97 @@ void TasmanianTridiagonalSolver::decompose(int n, std::vector<double> &d, std::v
     }
 }
 
+void TasmanianTridiagonalSolver::decompose2(std::vector<double> &diag, std::vector<double> &off_diag, const double mu0,
+                                            std::vector<double> &nodes, std::vector<double> &weights) {
+
+    // Ensure compatibility with the ALGOL implementation.
+    // NOTE: diag and off_diag are 1-indexed, while nodes and weights are 0-indexed after this step.
```

Comment:
This is beyond confusing, use C++ style of 0-indexing everywhere. It may be hard to read the Fortran code, but you can adjust the indexing.

Reply:
I'm not sure if I want to do this. The ALGOL code does both 1-indexing and 0-indexing in some places because it stores an additional variable in some arrays. 

For example, in `diag`, the array contains `n+1` elements, where the last `n` elements are the actually diagonal elements and the first element is an auxiliary one used in the QR computation, but which needs to be in position `0`. In C++, I would need to somehow store that element in "position" `-1`.

Reply:
This is confusing. You can leave it for now, but I want to fix it eventually.